packages/components/src/DataTable/DataTable.tsx

```diff
@@ -0,0 +1,279 @@
+                                className: header.column.getCanSort()
+                                  ? "cursor-pointer select-none"
+                                  : "",
+                                onClick:
+                                  header.column.getToggleSortingHandler(),
+                              }}
+                            >
+                              {flexRender(
+                                header.column.columnDef.header,
+                                header.getContext(),
+                              )}
+                              {{
+                                asc: <Icon name="longArrowUp" color="green" />,
+                                desc: (
+                                  <Icon name="longArrowDown" color="green" />
+                                ),
+                              }[header.column.getIsSorted() as string] ?? null}
+                            </div>
+                          ) : (
+                            <div>
+                              {flexRender(
+                                header.column.columnDef.header,
+                                header.getContext(),
+                              )}
+                            </div>
+                          )}
+                        </>
+                      )}
+                    </th>
+                  );
+                })}
+              </tr>
+            ))}
+          </thead>
+          <tbody>
+            {table.getRowModel().rows.map(row => {
+              return (
+                <tr key={row.id}>
+                  {row.getVisibleCells().map(cell => {
+                    return (
+                      <td key={cell.id}>
+                        {flexRender(
+                          cell.column.columnDef.cell,
+                          cell.getContext(),
+                        )}
+                      </td>
+                    );
+                  })}
+                </tr>
+              );
+            })}
+          </tbody>
+        </table>
+      </div>
+      {pagination && (
+        <Footer table={table} itemsPerPage={pagination.itemsPerPage} />
+      )}
+    </div>
+  );
+}
+
+function createTableSettings<T>(
+  data: T[],
+  columns: ColumnDef<T>[],
+  options?: { pagination?: Pagination; sorting?: Sorting },
+) {
+  let tableSettings: TableOptions<T> = {
+    data,
+    columns,
+    getCoreRowModel: getCoreRowModel(),
+  };
+
+  tableSettings = addPaginationSettings(tableSettings, options?.pagination);
+
+  tableSettings = addSortingSettings(tableSettings, options?.sorting);
+
+  return tableSettings;
+}
+
+function addPaginationSettings<T>(
+  tableSettings: TableOptions<T>,
+  pagination?: Pagination,
+) {
+  if (!pagination) return tableSettings;
+
+  const {
+    state: paginationState,
+    onPaginationChange,
+    manualPagination,
+  } = pagination;
+
+  const newTableSettings = { ...tableSettings }; //code smell??
```

Comment:
@lmorgan29 you mean something like this? 

```ts
interface PaginationSettings<T>
  extends Pick<
    TableOptions<T>,
    | "pageCount"
    | "onPaginationChange"
    | "manualPagination"
    | "getPaginationRowModel"
  > {
  paginationState?: PaginationState;
}

function getPaginationSettings<T>(
  pagination?: Pagination
): PaginationSettings<T> {
  const paginationSettings: PaginationSettings<T> = {};

  if (!pagination) {
    return paginationSettings;
  }

  const { manualPagination, state, pageCount, onPaginationChange } = pagination;

  paginationSettings.manualPagination = manualPagination;

  if (manualPagination) {
    if (state) paginationSettings.paginationState = state;

    if (pageCount) paginationSettings.pageCount = pageCount;

    if (onPaginationChange) {
      paginationSettings.onPaginationChange = onPaginationChange;
    }
  } else {
    paginationSettings.getPaginationRowModel = getPaginationRowModel();
  }

  return paginationSettings;
}
```

Reply:
btw, when we tried to pass onPaginationChange, getPaginationRowModel, etc... directly to useReactTable it kinda broke the whole thing, so apparently, we cannot set some of the options as undefined even though their type says it can be undefined. 

that's why many ifs where the immediate solution.

Reply:
Yeah, I think your change makes this a bit cleaner!

> when we tried to pass onPaginationChange, getPaginationRowModel, etc... directly to useReactTable it kinda broke the whole thing
Weird 🤔  thanks for explaining that